src/lib/actions/pagination.ts

```diff
@@ -0,0 +1,36 @@
+import { Payload } from './type-alias';
+import { Type } from '@angular/core';
+import { EntityState } from '../entity-state';
+import { generateActionObject } from '../internal';
+
+export type GoToPagePayload =
```

Comment:
@markwhitfeld Do you have any idea how to simplify that? Maybe using conditional types?

Reply:
This looks great

Reply:
While I think that this version is fine, I also quickly wrote an alternative

```typescript
export class StepPagePayload implements Payload<{ step: number; wrap: boolean }> {
  private constructor(public readonly payload: { step: number; wrap: boolean }) {
  }

  static with(step: number, wrap: boolean = false) {
    return new StepPagePayload({step, wrap});
  }
}

export class NextPagePayload {
  static with(wrap: boolean = false) {
    return StepPagePayload.with(1, wrap);
  }
}

export class GoToPagePayload implements Payload<number> {
  private constructor(public readonly payload: number) {
  }

  static with(index: number) {
    return new GoToPagePayload(index);
  }
}

// etc.
```

But I think then there are too many classes etc. floating around. With the current version you also have strongly typed payloads and there are not even more class names to remember.